docs/README.md

```diff
@@ -12,3 +12,177 @@ We currently have the following components available for React applications:
 -   [Score Card](https://github.com/brightlayer-ui/react-component-library/blob/dev/docs/ScoreCard.md)
 -   [Spacer](https://github.com/brightlayer-ui/react-component-library/blob/dev/docs/Spacer.md)
 -   [User Menu](https://github.com/brightlayer-ui/react-component-library/blob/dev/docs/UserMenu.md)
+
+---
+
+## Style Overrides Guide
+
+Material UI, and by extension, Brightlayer UI provides several different ways to customize a component's styles. Your specific context will determine which one is ideal. From narrowest to broadest use case, here are the options:
+
+### 1. The `sx` prop
+
+The [sx prop](https://mui.com/system/basics/#the-sx-prop) is the best option for adding style overrides to a single instance of a component in most cases.
+
+#### Usage
+
+```tsx
+import { InfoListItem } from '@brightlayer-ui/react-components';
+...
+<InfoListItem
+    title={'Info List Item'}
+    sx={{ my: 2, backgroundColor: '#efefef' }}
+/>
+```
+
+### 2. Overriding nested component styles
+
+To customize a specific part of a component, you can use the global CSS class provided by Brightlayer UI inside the sx prop. These global CSS class names can be found in the `Style Overrides` section of the documentation for each component. As an example, let's say you want to customize the InfoListItem component's icon size and padding.
+
+#### Usage
+
+```tsx
+import { InfoListItem } from '@brightlayer-ui/react-components';
+import { Leaf } from '@brightlayer-ui/icons-mui';
+...
+<InfoListItem
+    title={'Info List Item'}
+    icon={<Leaf />}
+    sx={{ width: '100%', '& .BluiInfoListItem-icon' : { ml: '2px', fontSize: '1rem' } }}
+/>
+```
+
+### 3. Overriding styles with class names
+
+If you want to override a component's styles using custom classes, you can use the className prop, available on each component. To override the styles of a specific part of the component, use the global CSS classes provided, as described in the previous section "Overriding nested component styles".
+
+### 4. Styled component
+
+To reuse the same overrides in different locations across your application, create a reusable component using the [styled()](https://mui.com/system/styled/) utility.
+
+#### Usage
+
+```tsx
+import * as React from 'react';
+import { styled } from '@mui/material/styles';
+import { InfoListItem, InfoListItemProps } from '@brightlayer-ui/react-components';
+import { CheckCircle } from '@mui/icons-material';
+...
+const SuccessInfoListItem = styled(InfoListItem)<InfoListItemProps>(({ theme }) => ({
```

Comment:
This is _kind_ of a weak example (I think it probably makes more sense to just use the iconColor props instead of defining a styled component for this)

Reply:
The point wasn't to have a "great" example, but to have something that demonstrates the general usage, but I see your point. I'll update this.